pkgs/applications/networking/instant-messengers/whatsie/default.nix

```diff
@@ -0,0 +1,52 @@
+{
+  fetchFromGitHub,
+  lib,
+  stdenv,
+  xorg,
+  full,
+  qtwebengine,
+  wrapQtAppsHook,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "whatsie";
+  version = "4.15.3";
+
+  src = fetchFromGitHub {
+    owner = "keshavbhatt";
+    repo = "whatsie";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-fbbyh8Q1jbKY+4ZEexDvIrhdcP+uCYAnhtUQsSqS0jw=";
+  };
+
+  buildInputs = [
+    xorg.libX11
+    xorg.libxcb
+    full
+  ];
+  nativeBuildInputs = [ wrapQtAppsHook ];
+
+  configurePhase = ''
```

Comment:
Two things:

1. Usually we set `sourceRoot`, as in `sourceRoot = ${src.name}/src;`
1. Usually `qmake` goes into `nativeBuildInputs`, so that it is invoked automatically